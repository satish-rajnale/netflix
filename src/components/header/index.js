import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Profile,
  PlayButton,
  Picture,
  Dropdown,
  Group,
  Link,
  FeatureCallOut,
  Text,
  Background,
  ButtonLink,
  Container,
  Logo,
  Feature,
  SearchInput,
  SearchIcon,
  Search,
} from "./styles/header";
import React from 'react';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        placeholder="Search films & series"
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
      />
    </Search>
  );
};
