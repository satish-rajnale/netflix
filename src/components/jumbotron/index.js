import React from 'react';
import {Inner} from './styles/jumbotron'

export default function Jumbotron({children, direction = 'row', ...restProps}){
    return (
        <Inner direction={direction}>
            <p>Hello again</p>
        </Inner>
    )
}


Jumbotron.Container = function JumbotronContainer({ children, ...restProps}){
    return <Container {...restProps}>{children}</Container>
}