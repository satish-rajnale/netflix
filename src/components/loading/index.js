import { LockBody, Spinner, Picture, ReleaseBody } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleasebody() {
  return <ReleaseBody />;
};
