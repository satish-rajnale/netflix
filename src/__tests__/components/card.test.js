import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { Card, Player } from "../../components";
 
const category = "series";

const slideRows = [
  {
    title: "Documentaries",
    data: [
      {
        title: "Tiger King",
    description:
      "Tiger King description",
    genre: "documentaries",
    maturity: "18",
    slug: "tiger-king",
    id: "2fc20b4c-5877-4de1-8782-1fe3a0a826a4",
    docId: "wbIZxQNif6X9yGoiHgcj"
      },
    ],
  },
  {
    title: "Feel Good",
    data: [
      {
        title: "Juno",
    description:
      "Juno description",
    genre: "feel-good",
    maturity: "12",
    slug: "juno",
    id: "e4f94668-2b91-4fdb-8c6b-bede8c143280",
    docId: "UIJFp7ORsveL1RVEkJBn",
      },
    ],
  },
];


describe("<Card/>", ()=>{
    afterEach(cleanup)
    it("renders the <Card/> with data", () => {
        const {container, getByText} = render( <Card.Group>
            {slideRows.map((slideItem) => (
              <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
                <Card.Title>{slideItem.title}</Card.Title>
    
                <Card.Entities>
                  {slideItem.data.map((item) => (
                    <Card.Item key={item.docId} item={item}>
                      <Card.Image
                        src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                      />
                      <Card.Meta>
                        <Card.SubTitle>{item.title}</Card.SubTitle>
                        <Card.Text>{item.description}</Card.Text>
                      </Card.Meta>
                    </Card.Item>
                  ))}
                </Card.Entities>
                <Card.Feature category={category}>
                  <Player>
                    <Player.Button />
                    <Player.Video src="/videos/bunny.mp4" />
                  </Player>
                </Card.Feature>
              </Card>
            ))}
          </Card.Group>);

          expect(getByText("Documentaries")).toBeTruthy();
          expect(getByText("Tiger King")).toBeTruthy();
          expect(getByText("Tiger King description")).toBeTruthy();
          expect(getByText("Feel Good")).toBeTruthy();
          expect(getByText("Juno")).toBeTruthy();
          expect(getByText("Juno description")).toBeTruthy();
    
        expect(container.firstChild).toMatchSnapshot();
        });

        it("renders the <Card/> on click with data", () => {
            const {container, queryByText, getByTestId, getByAltText} = render( <Card.Group>
                {slideRows.map((slideItem) => (
                  <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
                    <Card.Title>{slideItem.title}</Card.Title>
        
                    <Card.Entities>
                      {slideItem.data.map((item) => (
                        <Card.Item key={item.docId} item={item} data-testid={`${item.slug}-item-feature`}>
                          <Card.Image
                            src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                          />
                          <Card.Meta>
                            <Card.SubTitle>{item.title}</Card.SubTitle>
                            <Card.Text>{item.description}</Card.Text>
                          </Card.Meta>
                        </Card.Item>
                      ))}
                    </Card.Entities>
                    <Card.Feature category={category}>
                      <Player>
                        <Player.Button />
                        <Player.Video src="/videos/bunny.mp4" />
                      </Player>
                    </Card.Feature>
                  </Card>
                ))} 
              </Card.Group>);
    
              expect(queryByText("18")).toBeFalsy();
             fireEvent.click(getByTestId("tiger-king-item-feature"));
             expect(queryByText("18")).toBeTruthy();
            
             fireEvent.click(getByAltText("close"));
             expect(queryByText("18")).toBeFalsy();
             
             expect(queryByText("PG")).toBeFalsy();
             fireEvent.click(getByTestId("tiger-king-item-feature"));
             expect(queryByText("PG")).toBeFalsy();

            //  fireEvent.click(getByAltText("close"));
            //  expect(queryByText("PG")).toBeFalsy();


            expect(container.firstChild).toMatchSnapshot();
            })
})