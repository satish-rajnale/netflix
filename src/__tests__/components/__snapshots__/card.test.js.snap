// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders the <Card/> on click with data 1`] = `
<div
  class="card__Group-s87lnr-2 ejogLi"
>
  <div
    class="card__Container-s87lnr-1 bGhkvJ"
  >
     
    <p
      class="card__Title-s87lnr-0 gMKuUI"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-s87lnr-5 bcGoX"
    >
      <div
        class="card__Item-s87lnr-8 huWLnU"
        data-testid="tiger-king-item-feature"
      >
        <img
          class="card__Image-s87lnr-7 ffNVYX"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-6 eBqRRD"
        >
          <p
            class="card__SubTitle-s87lnr-3 eMgeKg"
          >
            Tiger King
          </p>
          <p
            class="card__Text-s87lnr-4 gQaeVh"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
    <div
      class="card__Feature-s87lnr-10 dgatLU"
      src="/images/series/documentaries/tiger-king/large.jpg"
    >
      <div
        class="card__Content-s87lnr-13 kRLAES"
      >
        <p
          class="card__Title-s87lnr-0 card__FeatureTitle-s87lnr-11 gMKuUI ewDSEC"
        >
          Tiger King
        </p>
        <p
          class="card__FeatureText-s87lnr-9 clUgkC"
        >
          Tiger King description
        </p>
        <button
          class="card__FeatureClose-s87lnr-12 foVqdi"
        >
          <img
            alt="close"
            src="/images/icons/close.png"
          />
        </button>
        <div
          class="card__Group-s87lnr-2 iyjHSy"
        >
          <div
            class="card__Maturity-s87lnr-14 kFrKns"
          >
            18
          </div>
          <p
            class="card__FeatureText-s87lnr-9 hIJsAk"
            font-weight="bold"
          >
            Documentaries
          </p>
        </div>
        <div
          class="player__Container-sc-1c49yiu-0"
        >
          <button
            class="player__Button-sc-1c49yiu-4 liGXfw"
          >
            Play
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-s87lnr-1 bGhkvJ"
  >
     
    <p
      class="card__Title-s87lnr-0 gMKuUI"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-s87lnr-5 bcGoX"
    >
      <div
        class="card__Item-s87lnr-8 huWLnU"
        data-testid="juno-item-feature"
      >
        <img
          class="card__Image-s87lnr-7 ffNVYX"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-6 eBqRRD"
        >
          <p
            class="card__SubTitle-s87lnr-3 eMgeKg"
          >
            Juno
          </p>
          <p
            class="card__Text-s87lnr-4 gQaeVh"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card/> renders the <Card/> with data 1`] = `
<div
  class="card__Group-s87lnr-2 ejogLi"
>
  <div
    class="card__Container-s87lnr-1 bGhkvJ"
  >
     
    <p
      class="card__Title-s87lnr-0 gMKuUI"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-s87lnr-5 bcGoX"
    >
      <div
        class="card__Item-s87lnr-8 huWLnU"
      >
        <img
          class="card__Image-s87lnr-7 ffNVYX"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-6 eBqRRD"
        >
          <p
            class="card__SubTitle-s87lnr-3 eMgeKg"
          >
            Tiger King
          </p>
          <p
            class="card__Text-s87lnr-4 gQaeVh"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-s87lnr-1 bGhkvJ"
  >
     
    <p
      class="card__Title-s87lnr-0 gMKuUI"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-s87lnr-5 bcGoX"
    >
      <div
        class="card__Item-s87lnr-8 huWLnU"
      >
        <img
          class="card__Image-s87lnr-7 ffNVYX"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-s87lnr-6 eBqRRD"
        >
          <p
            class="card__SubTitle-s87lnr-3 eMgeKg"
          >
            Juno
          </p>
          <p
            class="card__Text-s87lnr-4 gQaeVh"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
